contracts/InvestorVesting.sol

```diff
@@ -0,0 +1,162 @@
+ *
+ * Predetermined agreement with investors for a linear unlock over three years, with a six month cliff
+ */
+
+ contract InvestorVesting is ReentrancyGuard, Governable {
+ 
+    /// @notice SOLACE Token.
+    address public solace;
+
+    /// @notice custodian of SOLACE tokens - EOA that transfers investor SOLACE tokens to this contract
+    address public custodian;
+
+    /// @notice timestamp that investor tokens start vesting.
+    uint256 public vestingStart;
+
+    /// @notice timestamp that cliff for investor tokens finishes.
+    uint256 public cliff;
+
+    /// @notice timestamp that investor tokens finish vesting.
+    uint256 public vestingEnd;
+ 
+    /// @notice Total tokens for an investor.
+    mapping(address => uint256) public totalInvestorTokens;
+
+    /// @notice Redeemed tokens for an investor.
+    mapping(address => uint256) public redeemedInvestorTokens;
+
+    // @dev used in our floating point price maths
+    // Specifically where we determine unlocked proportion of investor tokens
+    uint256 internal constant Q12 = 1e12;
+
+    /**
+     * @notice Constructs the `InvestorVesting` contract.
+     * @param governance_ The address of the [governor](/docs/protocol/governance).
+     * @param solace_ Address of [**SOLACE**](./SOLACE).
+     * @param custodian_ Address which will transfer investor SOLACE tokens to this contract
+     * @param vestingStart_ Unix timestamp for start of vesting period for investor tokens
+     */
+    constructor(address governance_, address solace_, address custodian_, uint256 vestingStart_) Governable(governance_) {
+        require(solace_ != address(0x0), "zero address solace");
+        require(custodian_ != address(0x0), "zero address custodian");
+        require(vestingStart_ != 0, "vestingStart cannot be initialized as 0");
+        solace = solace_;
+        custodian = custodian_;
+        vestingStart = vestingStart_;
+        cliff = vestingStart_ + 15768000; // Cliff is 6-months after vesting start
+        vestingEnd = vestingStart_ + 94608000; // Vesting ends 3-years after vesting start
+    }
+
+    /***************************************
+    VIEW FUNCTIONS
+    ***************************************/
+
+    /// @notice View total SOLACE token amount for an investor address.
+    function viewTotalTokens(address investor) external view returns (uint256) {
+        return totalInvestorTokens[investor];
+    }
+
+    /// @notice View redeemed SOLACE token amount for an investor address.
+    function viewRedeemedTokens(address investor) external view returns (uint256) {
+        return redeemedInvestorTokens[investor];
+    }
+
+    /// @notice View unlocked SOLACE token amount that an investor can claim 
+    function viewRedeemableUnlockedTokens(address investor) external view returns (uint256) {
+        uint256 redeemableUnlockedAmount = _getRedeemableUnlockedTokens(investor);
+        return redeemableUnlockedAmount;
+    }
```

Comment:
we can get rid of these since they're external wrappers on public variables or internal functions, then turn getRedeemableUnlockedTokens into a public function